kjdoc/tour.md

```diff
@@ -1021,7 +1021,24 @@ KJ provides an advanced, cross-platform filesystem API in `kj/filesystem.h`. Fea
 * Symlinks, hard links, listing directories, recursive delete, recursive create parents, recursive copy directory, memory mapping, and unnamed temporary files are all exposed and easy to use.
 * Sparse files ("hole punching"), copy-on-write file cloning (`FICLONE`, `FICLONERANGE`), `sendfile()`-based copying, `renameat2()` atomic replacements, and more will automatically be used when available.
 
+```c++
+// NOTE: in most cases, you should construct the `Filesystem` at the entry point of your program,
+// then pass it in as a parameter wherever it's used. This allows test implementations or user-space
+// filesystems.
+kj::Own<kj::Filesystem> fs = kj::newDiskFilesystem();
+kj::Own<const kj::ReadableFile>> file = fs->getCurrent().openFile(kj::Path::parse("README.md"));
+kj::String text = file->readAllText();
+```
+
+For large files, consider using mmap when you want to read the full contents of a file, which uses
+on demand paging as contents are accessed rather than buffering the contents into memory first.
+
+```c++
+kj::Array<const char> fileBuffer = file->mmap(0, file->stat().size).releaseAsChars();
+```
+
 See the API reference (header file) for details.
+See `parseFromDirectory` in `capnp/schema-parser.h` for more examples.
```

Comment:
This document is about KJ, which is intended to be independent of Cap'n Proto and may actually split into its own project someday, so I'm not sure I like referring to capnp source files like this. Also the examples there are intertwined with SchemaParser. If we need better examples for KJ, we should probably write them directly in `filesystem.h`.

Reply:
I'll add some more examples in filesystem.h.

Reply:
Sadly, userspace virtual filesystems are not a standard feature of operating systems. Technically on Linux there is FUSE but setting that up is complicated and requires privileges.

So yeah, the in-memory FS stuff is really meant for passing to KJ-based libraries. The most common use case would be unit tests, so that you can test code that does file manipulation without creating real files on disk.